lib/jobs/redfish-discovery.js

```diff
@@ -245,33 +246,67 @@ function RedfishDiscoveryJobFactory(
                 targetList.push(target);
             });
             
+            var identifiers = [];
             var config = Object.assign({}, self.settings);
             config.root = data.system['@odata.id'];
-            var node = {
-                type: 'compute',
-                name: data.system.Name,
-                identifiers: [ data.system.Id ],
-            };
             
-            var relations = [{
-                relationType: 'enclosedBy',
-                targets: targetList
-            }];
-
-            var obm = {
-                config: config,
-                service: 'redfish-obm-service'
-            };
-            
-            return self.upsertRelations(node, relations)
-            .then(function(nodeRecord) {
-                return Promise.all([
-                    waterline.obms.upsertByNode(nodeRecord.id, obm),
-                    nodeRecord
-                ]);
+            return self.redfish.clientRequest(config.root)
+            .then(function(res) {
+                var ethernet = _.get(res.body, 'EthernetInterfaces');
+                if(ethernet) {
+                    return self.redfish.clientRequest(ethernet['@odata.id'])
+                    .then(function(res) {
+                        assert.object(res, 'ethernet interfaces');
+                        return res.body.Members;
+                    })
+                    .map(function(intf) {
+                        return self.redfish.clientRequest(intf['@odata.id'])
+                        .then(function(port) {
+                            assert.object(port, 'ethernet port');
+                            if(_.has(port.body, 'MACAddress')) {
+                                identifiers.push(port.body.MACAddress.toLowerCase()); 
+                            };
+                        }); 
+                    })
+                    .catch(function(err) {
+                        logger.error(
+                            'Error gathering ethernet information from System',
+                            { error: err, root: config.root }
+                        );
+                        return; // don't hold up the other system resources
+                    });
+                }
             })
-            .spread(function(obm, node) {
-                return node;
+            .then(function() {
+                if(_.isEmpty(identifiers)) {
+                    identifiers.push(data.system.Id);
+                }
+                var node = {
+                    type: 'compute',
+                    name: data.system.Name,
+                    identifiers: identifiers
+                };
+                
+                var relations = [{
+                    relationType: 'enclosedBy',
+                    targets: targetList
+                }];
+
+                var obm = {
+                    config: config,
```

Comment:
A lot properties in the `config`, are they all necessary for redfish-obm-service? If not all, please only pick those necessary properties before upsert the obm.